x509util/certcheck/certcheck.go

```diff
@@ -133,33 +141,35 @@ func chainFromSite(target string) ([]*x509.Certificate, error) {
 	goChain := conn.ConnectionState().PeerCertificates
 	chain := make([]*x509.Certificate, len(goChain))
 	for i, goCert := range goChain {
-		cert, err := x509.ParseCertificate(goCert.Raw)
-		if err != nil {
+		var cert *x509.Certificate
+		cert, err = x509.ParseCertificate(goCert.Raw)
+		if x509.IsFatal(err) {
 			return nil, fmt.Errorf("%s: failed to convert Go Certificate [%d]: %v", target, i, err)
 		}
 		chain[i] = cert
 	}
 
-	return chain, nil
+	return chain, err
```

Comment:
`err` here will only be the error returned by the last call to `x509.ParseCertificate()`.

Reply:
True; added a TODO for now

Reply:
Is this really deferrable? It seems like a bug. If the last call to `ParseCertificate()` returns a nil err, then so too will `chainFromSite()`, even if all previous calls to `ParseCertificate()` returned a non-fatal error. As a result, with the --strict flag, you'll end up exiting with code 0 even though non-fatal errors occurred.

Reply:
Ah, I misunderstood what you meant -- yeah, that's wrong, I'll fix it.